pmd-java/src/main/java/net/sourceforge/pmd/lang/java/oom/signature/OperationSignature.java

```diff
@@ -137,62 +123,28 @@ private static boolean isGetterOrSetter(ASTMethodDeclaration node) {
             return isGetter(node, fieldNames) || isSetter(node, fieldNames);
         }
 
+
         /** Attempts to determine if the method is a getter. */
         private static boolean isGetter(ASTMethodDeclaration node, Map<String, String> fieldNames) {
 
-
-            List<ASTReturnStatement> returnStatements
-                = node.getBlock().findDescendantsOfType(ASTReturnStatement.class);
-
-            for (ASTReturnStatement st : returnStatements) {
-                ASTName name = st.getFirstDescendantOfType(ASTName.class);
-                if (name == null) {
-                    continue;
-                }
-
-                if (fieldNames.containsKey(name.getImage().split("\\.")[0])) {
-                    return true;
-                }
+            if (node.getFirstDescendantOfType(ASTFormalParameters.class).getParameterCount() != 0
+                || node.getFirstDescendantOfType(ASTResultType.class).isVoid()) {
+                return false;
             }
+            return fieldNames.containsKey(node.getName());
```

Comment:
Not sure, if this works, since node.getName() is e.g. `getAge`, and fieldname is `age`...